### PR TITLE
UI for webhooks

### PR DESCRIPTION
**Pull request by sei-byoung**
Implement a UI to view and manage webhook subscriptions

### CODE REVIEW

File: src/app/components/admin-app/app-admin-subscription-search/app-admin-subscription-search.component.ts [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+  addNewSubscription() {
+    this.dialogService.editSubscription().subscribe(resp => {
+      console.log(resp);
+      this.refreshSubs();
+    })
+  }
+
+  refreshSubs() {
+    this.webhookService.getAll().subscribe((subs) => {
```
Comment (sei-tspencer) [inline src/app/components/admin-app/app-admin-subscription-search/app-admin-subscription-search.component.ts:48]: ".subscribe" should be replaced by ".pipe(takeUntil(this.unsubscribe$)).subscribe" and an ngOnDestroy() function should be added.  An example can be seen in logged-in-user.service.ts line 42 and following or app.component.ts lines 30 and 153.

File: src/app/components/admin-app/app-admin-subscription-search/edit-subscription/edit-subscription.component.ts [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+  onSubmit() {
+    const webhook = <WebhookSubscriptionForm> {
+      name: this.form.get('name').value as string,
+      callbackUri: this.form.get('callback').value as string,
+      clientId: this.form.get('clientId').value as string,
+      clientSecret: this.form.get('clientSecret').value as string,
+      eventTypes: this.form.get('events').value as EventType[]
+    };
+
+    this.webhookService.subscribe(webhook).subscribe(
```
Comment (sei-tspencer) [inline src/app/components/admin-app/app-admin-subscription-search/edit-subscription/edit-subscription.component.ts:50]: replace ".subscribe" with ".pipe(take(1)).subscribe".  No ngOnDestroy needed for take(1)

File: src/app/components/admin-app/app-admin-subscription-search/edit-subscription/edit-subscription.component.ts [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+        this.dialogRef.close(false);
+      },
+      err => {
+        this.dialogRef.close(true);
+      }
+    );
+  }
+
+  onDelete() {
+    this.webhookService._delete(this.currentSubId).subscribe(
```
Comment (sei-tspencer) [inline src/app/components/admin-app/app-admin-subscription-search/edit-subscription/edit-subscription.component.ts:61]: replace ".subscribe" with ".pipe(take(1)).subscribe".  No ngOnDestroy needed for take(1)

**Comment by sei-aschlackman** [comment]
Superseded by #469

**Pull request closed without merging**